staker/block_validator.go

```diff
@@ -327,6 +351,27 @@ func NewBlockValidator(
 			PosInBatch: 0,
 		}
 	}
+	if config().Dangerous.Revalidation.StartBlock > 0 {
+		startBlock := config().Dangerous.Revalidation.StartBlock
+		messageCount, err := inbox.GetBatchMessageCount(startBlock - 1)
+		if err != nil {
+			return nil, err
+		}
+		res, err := streamer.ResultAtMessageIndex(messageCount)
+		if err != nil {
+			return nil, err
+		}
+		gs := validator.GoGlobalState{
+			BlockHash:  res.BlockHash,
+			SendRoot:   res.SendRoot,
+			Batch:      startBlock,
+			PosInBatch: 0,
```

Comment:
obs: I don't have much context to understand why we use those values here.

Reply:
Actually i mistook batch for block, updated it now.